src/com/nicoles_change_maker/ChangeMaker.java

```diff
@@ -0,0 +1,61 @@
+package com.nicoles_change_maker;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
+/**
+ * Created by bears8yourface on 1/5/16.
+ */
+public class ChangeMaker {
+    double remainingDollars;
+
+    public HashMap makeChange(double dollarAmount) {
+        remainingDollars = dollarAmount;
+
+        if (checkDollarAmountIsValid(dollarAmount)) {
+            return evaluateEachCoinAmount();
+        }
+        throw new IllegalArgumentException();
+    }
+
+    private HashMap evaluateEachCoinAmount() {
+        HashMap changeCollection = new HashMap();
+        Iterator coinIterator = new CoinMap().getIterator();
+
+        while(coinIterator.hasNext()) {
+            Map.Entry coinValueMapping = (Map.Entry) coinIterator.next();
+            evaluateCoinAmountForInclusionInChangeCollection(changeCollection, coinValueMapping);
+        }
+
+        return changeCollection;
+    }
+
+    private void evaluateCoinAmountForInclusionInChangeCollection(HashMap changeCollection, Map.Entry coinValueMapping) {
+        String coinType = (String) coinValueMapping.getKey();
```

Comment:
Here's another spot where adding type parameters to generics comes in handy: You shouldn't need to cast these values to `String` and `Double`, because the compiler will already be sure that `coinValueMapping` keys are strings and values are doubles.